app/src/main/java/com/android/sample/ui/listActivities/ViewListActivities.kt

```diff
@@ -84,6 +96,26 @@ fun ListActivitiesScreen(
   val profile = profileViewModel.userState.collectAsState().value
   var searchText by remember { mutableStateOf("") }
 
+  val locationPermissionLauncher =
+      rememberLauncherForActivityResult(
+          contract = ActivityResultContracts.RequestPermission(),
+          onResult = { isGranted ->
+            if (isGranted) {
+              locationViewModel.fetchCurrentLocation()
+            } else {
+              Log.d("OverviewScreen", "Location permission denied by the user.")
+            }
+          })
+
+  LaunchedEffect(Unit) {
+    if (ContextCompat.checkSelfPermission(context, Manifest.permission.ACCESS_FINE_LOCATION) ==
+        PackageManager.PERMISSION_GRANTED) {
+      locationViewModel.fetchCurrentLocation()
+    } else {
+      locationPermissionLauncher.launch(Manifest.permission.ACCESS_FINE_LOCATION)
+    }
+  }
+
```

Comment:
This adds lots of code lines. I don't know if there is a more graceful way to check for this, as it is purely duplicated code from `ui/map/Map.kt` 
Also, this part is not tested (and is actually hard to test).
But it is great that we do not need to go through maps to load the distances.

I would recommend this permission grant is rather demanded when the app launches.